### PR TITLE
Additional columns

### PR DESCRIPTION
**Pull request by adrianreber**
If MM2 should go productive soon I would like to add additional fields to the database so that additional crawler features do not need changes to the database.

### CODE REVIEW

File: mirrormanager2/lib/model.py [hunk 1]
```diff
@@ -193,6 +193,24 @@ class Host(BASE):
...
+    # This can be used to auto disable a host if the crawler fails
+    # multiple times in a row.
+    crawl_failures = sa.Column(sa.Integer, default=0, nullable=False)
+    # Add a text field to specify why the mirror was disabled.
+    # This can either be filled by the crawler for auto disable due
+    # to crawl failures, or by an admin (e.g., ticket number)
+    disable_reason = sa.Column(sa.Text(), nullable=True)
+    # If SSH based push mirroring will ever be implemented
+    # this field should contain the public key of the destination host
+    push_ssh_public_key = sa.Column(sa.Text(), nullable=True)
```
Comment (pypingou) [inline mirrormanager2/lib/model.py]: Don't we need the private key if we want to push there?

**Comment by pypingou** [comment]
There is a ``..pyo`` file in there that could be dropped I think.

**Comment by adrianreber** [comment]
Yes, private key, of course. I will update this PR.

There is also a .pyc file. Can that also be dropped?

**Comment by pypingou** [comment]
I think so for the ``.pyc`` yes

**Comment by adrianreber** [comment]
PR udpated

File: alembic/versions/24681dabe5fa_add_host_columns.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+down_revision = None
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.add_column('host', sa.Column('crawl_failures', sa.Integer))
+    op.add_column('host', sa.Column('disable_reason', sa.Text()))
+    op.add_column('host', sa.Column('push_ssh_public_key', sa.Text()))
```
Comment (pypingou) [inline alembic/versions/24681dabe5fa_add_host_columns.py]: That one still says public_key

File: alembic/versions/24681dabe5fa_add_host_columns.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    op.add_column('host', sa.Column('push_ssh_host', sa.Text()))
+    op.add_column('host', sa.Column('push_ssh_command', sa.Text()))
+    op.add_column('host', sa.Column('last_crawls', sa.PickleType()))
+
+def downgrade():
+     op.drop_column('host', 'crawl_failures')
+     op.drop_column('host', 'disable_reason')
+     op.drop_column('host', 'push_ssh_public_key')
```
Comment (pypingou) [inline alembic/versions/24681dabe5fa_add_host_columns.py]: And here as well :)

**Comment by adrianreber** [comment]
Thanks for spotting these. Updated.

**Comment by pypingou** [comment]
:+1: for me

**Pull request merged**